how2meet/ui/pages/events.py

```diff
@@ -0,0 +1,139 @@
+from datetime import datetime
+import uuid
+from collections import namedtuple
+
+from nicegui import ui, APIRouter
+import httpx
+
+from ..components.frames import frame
+
+router = APIRouter(prefix="/events",
+                   tags=["events"])
+
+@router.page("/")
+async def events(user: str = None):
+    """List page for all events"""
+    with frame("Events"):
+        # Get the list of events from the API
+        # NOTE: This is a blocking call, so we use an async client
+        async with httpx.AsyncClient() as client:
+            events = await client.get(f"http://localhost:8000/api/events/", timeout=10)
+        try:
+            events = events.json()
+        except:
+            events = []
+
+        async def delete_event(event_id: str):
+            async with httpx.AsyncClient() as client:
+                response = await client.delete(f"http://localhost:8000/api/events/{event_id}", timeout=10)
+                response.raise_for_status()
+            ui.open("/events")
+
+        # Display the list of events
+        for event in events:
+            with ui.card().classes("w-full"):
+                with ui.row().classes("w-full justify-between"):
+                    with ui.column().classes("flex-grow"):
+                        ui.label(f"Event ID: {event['id']}")
+                        ui.label(f"Event Name: {event['name']}")
+                    ui.button("", icon="info", on_click=lambda event_id=event['id']: ui.open(f"/events/{event_id}")).classes("w-6 h-6")
+                    ui.button("", icon="delete", color="red", on_click=lambda event_id=event['id']: delete_event(event_id)).classes("w-6 h-6")
+
+        # Add navigation buttons
+        with ui.row().classes("w-full justify-center"):
+            ui.button("Back", on_click=lambda: ui.open("/"))
+            ui.button("New Event", on_click=lambda: ui.open(f"/events/new/{uuid.uuid4()}"))
+
+
+@router.page("/{event_id}")
+async def event_home(event_id: str):
+    """Detail page for a specific event"""
+    # Get the event from the API
+    # NOTE: This is a blocking call, so we use an async client
+
+    with frame("Event Home"):
+        async with httpx.AsyncClient() as client:
+            event = await client.get(f"http://localhost:8000/api/events/{event_id}", timeout=10)
+        try:
+            event = event.json()
+        except:
+            event = {}
+
+        # Display the event details
+        with ui.column().classes("w-full justify-between"):
+            for key, value in event.items():
+                ui.label(f"{key}: {value}")
+        ui.button("Back", on_click=lambda: ui.open("/events"))
+
+@router.page("/new/{event_id}")
+async def new_event(event_id: str):
+    """New event creation page
+
+    Args:
+        event_id (str): The unique ID of the event to be created.
+
+    Returns:
+        None
+    """
+
+
+    async def add_guest():
+        invite_uuid = str(uuid.uuid4())
```

Comment:
Related to previous comment--I'm not sure what we get out of casting the UUID to a string. Pedantic/FastAPI should be capable of data validation on non-standard types.

Reply:
gross